hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleContext.java

```diff
@@ -114,21 +132,32 @@ enum TransactionCategory {
      * there will be no corresponding {@link SignatureVerification}. If the key was provided during pre-handle, then the
      * corresponding {@link SignatureVerification} will be returned with the result of that verification operation.
      *
+     * <p>The signatures of required keys are guaranteed to be verified. Optional signatures may still be in the
+     * process of being verified (and therefore may time out). The timeout can be configured via the configuration
+     * {@code hedera.workflow.verificationTimeoutMS}
+     *
      * @param key the key to get the verification for
-     * @return the verification for the given key, or {@code null} if no such key was provided during pre-handle
+     * @return the verification for the given key
      * @throws NullPointerException if {@code key} is {@code null}
+     * @throws TimeoutException if the verification for the given key timed out
      */
-    @Nullable
-    SignatureVerification verificationFor(@NonNull Key key);
+    @NonNull
+    SignatureVerification verificationFor(@NonNull Key key) throws TimeoutException;
 
     /**
      * Gets the {@link SignatureVerification} for the given hollow account.
      *
+     * <p>The signatures of required accounts are guaranteed to be verified. Optional accounts may still be in the
+     * process of being verified (and therefore may time out). The timeout can be configured via the configuration
+     * {@code hedera.workflow.verificationTimeoutMS}
+     *
      * @param evmAlias The evm alias to lookup verification for.
      * @return the verification for the given hollow account.
+     * @throws NullPointerException if {@code evmAlias} is {@code null}
+     * @throws TimeoutException if the verification for the given alias timed out
      */
-    @Nullable
-    SignatureVerification verificationFor(@NonNull final Bytes evmAlias);
+    @NonNull
+    SignatureVerification verificationFor(@NonNull final Bytes evmAlias) throws TimeoutException;
```

Comment:
I would only differentiate if there is a good reason to do so because otherwise, we have to define how the complex keys should behave. 

If signatures are missing in a key list, does the key list fail, or is it missing? How should threshold keys behave? For instance, if there is a threshold key with 10 keys and a threshold of 8. Should it fail or miss if 3 of the keys are missing? What should the result be if 2 of them are missing? We can avoid these questions if we define missing and failing as equal.

Reply:
As discussed, we want to return a failing `SignatureVerification` if it is missing.